cpp/src/copying/get_element.cu

```diff
@@ -125,24 +125,35 @@ struct get_element_functor {
     CUDF_FAIL("get_element_functor not supported for list_view");
   }
 
-  template <typename T, std::enable_if_t<std::is_same<T, numeric::decimal32>::value> *p = nullptr>
+  template <typename T, std::enable_if_t<cudf::is_fixed_point<T>()> *p = nullptr>
   std::unique_ptr<scalar> operator()(
     column_view const &input,
     size_type index,
     rmm::cuda_stream_view stream,
     rmm::mr::device_memory_resource *mr = rmm::mr::get_current_device_resource())
   {
-    CUDF_FAIL("get_element_functor not supported for decimal32");
-  }
+    using Type = typename T::rep;
 
-  template <typename T, std::enable_if_t<std::is_same<T, numeric::decimal64>::value> *p = nullptr>
-  std::unique_ptr<scalar> operator()(
-    column_view const &input,
-    size_type index,
-    rmm::cuda_stream_view stream,
-    rmm::mr::device_memory_resource *mr = rmm::mr::get_current_device_resource())
-  {
-    CUDF_FAIL("get_element_functor not supported for decimal64");
+    auto device_col = column_device_view::create(input, stream);
+
+    rmm::device_scalar<Type> temp_data;
```

Comment:
Wish there was a smoother way to initialize a device scalar entirely on the gpu (without needing the gpu->cpu->gpu round trip for the validity)

Reply:
Good observation. We could add an overload of `rmm::device_vector::set_value()` that takes a pointer, and then use `cudaMemcpyAsync(..., cudaMemcpyDefault, stream)` to let CUDA handle figuring out where the pointer resides and do the copy. 

However, in the present usage, two calls to `set_value()` would require two separate (tiny) copies, whereas the `device_single_thread` below launches a single kernel.

Reply:
how about adding a constructor to all scalars that takes argument `rmm::device_scalar<bool>&& valid` ?